src/main/scala/mesosphere/marathon/state/Timestamp.scala

```diff
@@ -14,72 +14,71 @@ import scala.math.Ordered
 /**
   * An ordered wrapper for UTC timestamps.
   */
-abstract case class Timestamp private (private val utcDateTime: DateTime) extends Ordered[Timestamp] {
+abstract case class Timestamp private (private val instant: Instant) extends Ordered[Timestamp] {
   def toOffsetDateTime: OffsetDateTime =
     OffsetDateTime.ofInstant(
-      Instant.ofEpochMilli(utcDateTime.toInstant.getMillis),
-      utcDateTime.getZone.toTimeZone.toZoneId)
+      instant,
+      ZoneOffset.UTC)
 
-  def compare(that: Timestamp): Int = this.utcDateTime compareTo that.utcDateTime
+  def compare(that: Timestamp): Int = this.instant compareTo that.instant
 
-  def before(that: Timestamp): Boolean = (this.utcDateTime compareTo that.utcDateTime) < 0
-  def after(that: Timestamp): Boolean = (this.utcDateTime compareTo that.utcDateTime) > 0
+  def before(that: Timestamp): Boolean = (this.instant compareTo that.instant) < 0
+  def after(that: Timestamp): Boolean = (this.instant compareTo that.instant) > 0
   def youngerThan(that: Timestamp): Boolean = this.after(that)
   def olderThan(that: Timestamp): Boolean = this.before(that)
 
-  override def toString: String = utcDateTime.toString
+  override def toString: String = Timestamp.formatter.format(instant)
 
-  def toDateTime: DateTime = utcDateTime
+  def toInstant: Instant = instant
 
-  def millis: Long = toDateTime.getMillis
+  def millis: Long = toInstant.toEpochMilli
   def micros: Long = TimeUnit.MILLISECONDS.toMicros(millis)
   def nanos: Long = TimeUnit.MILLISECONDS.toNanos(millis)
 
   def until(other: Timestamp): FiniteDuration = {
-    val millis = other.utcDateTime.getMillis - utcDateTime.getMillis
-    FiniteDuration(millis, TimeUnit.MILLISECONDS)
+    val duration = Duration.between(this.instant, other.instant)
+    FiniteDuration(duration.toMillis, TimeUnit.MILLISECONDS)
   }
 
   /**
     * @return true if this timestamp is more than "by" duration older than other timestamp.
     */
   def expired(other: Timestamp, by: FiniteDuration): Boolean = this.until(other) > by
 
-  def +(duration: FiniteDuration): Timestamp = Timestamp(utcDateTime.getMillis + duration.toMillis)
-  def -(duration: FiniteDuration): Timestamp = Timestamp(utcDateTime.getMillis - duration.toMillis)
+  def +(duration: FiniteDuration): Timestamp = Timestamp(instant.plusMillis(duration.toMillis))
```

Comment:
you can do just instant.plus(duration) right?

Reply:
+1

Reply:
Unfortunately there is no such method.

Reply:
ah, we are not using java.time.Duration... ok then...